src/ctapipe/core/tests/test_tool.py

```diff
@@ -477,15 +479,33 @@ def start(self):
 def test_activity(tmp_path):
     """check that the config is correctly in the provenance"""
 
+    class Comp(Component):
+        quantity = traits.AstroQuantity(u.m).tag(config=True)
+
     class MyTool(Tool):
         name = "test_prov_log"
         description = "test"
-        userparam = Float(5.0, help="parameter").tag(config=True)
 
-    tool = MyTool()
+        value = Float(5.0, help="parameter").tag(config=True)
+        path = traits.Path(default_value=None, allow_none=True).tag(config=True)
+        time = traits.AstroTime(default_value=None, allow_none=True).tag(config=True)
 
+        def setup(self):
+            self.comp = Comp(parent=self)
+
+    tool = MyTool()
+    config = {
+        "MyTool": {
+            "value": 10.0,
+            "path": str(tmp_path / "foo.txt"),
+            "time": "2025-01-01T00:00:00",
+            "Comp": {
+                "quantity": {"value": 5.0, "unit": "m"},
+            },
```

Comment:
let's add also a TelescopeParameter to test, to make sure that we can correctly serialize UserList of the AstroQuantities

Reply:
done